### PR TITLE
[PyTorch] Sync loss with reduction='none' and L2Loss behaviour

### PR DESCRIPTION
**Pull request by AnirudhDagar**
Fixes logical error with mse loss being divided by number of elements which was wrong before.
Also make MSELoss behave like MXNet L2Loss.
Note: I've also tweaked the learning rate on purpose for PyTorch to improve the plots.

### CODE REVIEW

**Comment by astonzhang** [comment]
TODO@astonzhang: update d2l lib with changes in this PR

**Comment by astonzhang** [comment]
TODO: @astonzhang  translate English text into Chinese

**Comment by AnirudhDagar** [comment]
This is ready now. Except I don't know how to add this [commit](https://github.com/d2l-ai/d2l-en/pull/1984/commits/95eec3f6222d41dd5ef835b8115c007899874257) in Chinese. Can you add that after merging? I can't write Chinese. Thanks!

**Comment by mli** [comment]
Job d2l-zh/PR-1037/10 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-1037/
